### PR TITLE
Issue #13345: Enable examples tests for RecordTypeParameterNameCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for RecordTypeParameterNameCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **RecordTypeParameterNameCheck** in the project. The implementation focuses on ensuring that RecordTypeParameterName checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordtypeparametername/Example1.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+*/
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.recordtypeparametername;
+
+// xdoc section -- start
+class Example1 {
+  record MyRecord1<T>() {}
+
+  record MyRecord2<t>() {} // violation,
+  // the record type parameter name should match the regular expression "^[A-Z]$"
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordtypeparametername/Example1.java:16]: Please use `// violation below ` for all cases. You can always cut message at the beginning or end to satisfy line length

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordtypeparametername/Example1.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordtypeparametername/Example1.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
Thanks a lot !!

**Pull request merged**